addons/l10n_id/models/res_bank.py

```diff
@@ -76,3 +101,35 @@ def _get_qr_code_generation_params(self, qr_method, amount, currency, debtor_par
                 'value': self._get_qr_vals(qr_method, amount, currency, debtor_partner, free_communication, structured_communication),
             }
         return super()._get_qr_code_generation_params(qr_method, amount, currency, debtor_partner, free_communication, structured_communication)
+
+    def _l10n_id_qris_fetch_status(self, qr_data):
+        """
+        using self and the given data, fetches the status of a specific QR code generated by QRIS
+        Expected values in the qr_data dict are:
+            - invoice_id returned when generating a QR code
+            - the amount present in the qr code
+            - the datetime at which the QR code was generated
+        """
+        return self._l10n_id_make_qris_request('checkpaid_qris.php', {
+            'do': 'checkStatus',
+            'apikey': self.l10n_id_qris_api_key,
+            'mID': self.l10n_id_qris_mid,
+            'invid': qr_data['qris_invoice_id'],
+            'trxvalue': qr_data['qris_amount'],
+            'trxdate': qr_data['qris_creation_datetime'],
+        })
+
+    @api.model
+    def _l10n_id_make_qris_request(self, endpoint, params):
+        """ Make an API request to QRIS, using the given path and params. """
+        url = urljoin('https://qris.online/restapi/qris/', endpoint)
+        try:
+            response = requests.get(url, params=params, timeout=QRIS_TIMEOUT)
+            response.raise_for_status()
+            response = response.json()
+        except requests.exceptions.HTTPError as err:
+            raise ValidationError(_("Communication with QRIS failed. QRIS returned with the following error: %s", err))
+        except (requests.RequestException, ValueError):
+            raise ValidationError(_("Could not establish a connection to the QRIS API."))
+
+        return response
```

Comment:
@odoo/rd-security Maybe you would prefer to have this as a helper outside of a `Model` in order to be sure that it is not even accessible from a server action?

Reply:
@william-andre good idea, better safe than sorry